lib/authlogic/session/base.rb

```diff
@@ -1849,6 +1881,34 @@ def params_key
         build_key(self.class.params_key)
       end
 
+      def headers_credentials
+        controller.headers[headers_key]
+      end
+
+      def headers_enabled?
+        if !headers_credentials || !klass.column_names.include?("single_access_token")
+          return false
+        end
+        if controller.responds_to_single_access_allowed?
+          return controller.single_access_allowed?
+        end
+        headers_enabled_by_allowed_request_types?
+      end
+
+      def headers_enabled_by_allowed_request_types?
+        case single_access_allowed_request_types
+        when Array
+          single_access_allowed_request_types.include?(controller.request_content_type) ||
+            single_access_allowed_request_types.include?(:all)
+        else
+          %i[all any].include?(single_access_allowed_request_types)
+        end
+      end
```

Comment:
It seems `headers_enabled_by_allowed_request_types?` and `params_enabled_by_allowed_request_types?` have exactly the same contents. Let's fix that duplication by having a single method. What do you think about a name like `request_type_allowed?`, or `single_access_token_allowed_by_request_type?`?